packages/superset-ui-connection/src/types.ts

```diff
@@ -49,25 +43,21 @@ export interface RequestBase {
   method?: Method;
   parseMethod?: ParseMethod;
   postPayload?: PostPayload;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  json?: any;
   signal?: Signal;
   stringify?: Stringify;
   timeout?: ClientTimeout;
 }
 
-export interface RequestWithEndpoint extends RequestBase {
-  endpoint: Endpoint;
-  url?: undefined;
-}
-
-export interface RequestWithUrl extends RequestBase {
+export interface CallApi extends RequestConfig {
   url: Url;
-  endpoint?: undefined;
+  cache?: Cache;
+  redirect?: Redirect;
 }
 
-export type RequestConfig = RequestWithEndpoint | RequestWithUrl;
-
 export interface JsonTextResponse {
-  json?: Json;
+  json?: JsonObject;
```

Comment:
Hmmm `JsonObject` vs `JSONObject` seem easy to make a mistake.

Reply:
I think it's fine. In an idea world, we shouldn't need `JsonObject` if microsoft/TypeScript#15300 was fixed so these two should be interchangeable for most cases---so it's fine for users to use either. I just added aliases for `JsonValue`, `JsonArray`, etc., to make it more complete.

We can probably remove `JSONObject` but that would also remove the option of stricter typing.

Reply:
> I just added aliases for JsonValue, JsonArray, etc., to make it more complete.
Are they meant to be used anywhere? If not perhaps we can remove?

For `JsonObject`, we could rename that to `PlainObject`?

Reply:
It's different than `PlainObject`. The values shouldn't really be `any` if we had a choice (e.g. `new Person()` may be valid for `PlainObject`, but isn't valid for a JSON object parsed from `JSON.parse`). I think it makes sense to have a special name for this in case microsoft/TypeScript#15300 got fixed.

It also goes better with type names like `JsonResponse`, `TextResponse`, better.

Reply:
Here is my ask. I am OK with keeping the two types but would appreciate help reducing the confusion when to use `JSONObject` and when to use `JsonObject`. 
* The current names are not helpful. 
  * Maybe `JSONObject => StrictJsonObject`
  * or `JSONObject => JsonObject` and `JsonObject => RelaxedJsonObject` ?
* Add `/** */` comment for the types so autocomplete can help explain. such as

```
/**
 * If this fails to satisfy TypeScript when using as function arguments, use `RelaxedJsonObject` instead. 
 * `JsonObject` does not accept specific types when used as function arguments,
 * (Ref: https://github.com/microsoft/TypeScript/issues/15300).
 */
type JsonObject
/** 
 * This is less strict type of `JsonObject`. If that type fails to satisfy TypeScript when using as function arguments, use this type instead. 
 * `JsonObject` does not accept specific types when used as function arguments,
 * (Ref: https://github.com/microsoft/TypeScript/issues/15300).
 */
type RelaxedJsonObject
```

Reply:
Great suggestions! Happy to update!